### PR TITLE
Add global_ptr and local_ptr to address_space.hpp

### PR DESCRIPTION
**Pull request by JoanThibault**
PR relative to issue #20

### CODE REVIEW

File: tests/address_spaces/address_spaces.cpp [hunk 1]
```diff
@@ -18,6 +18,20 @@
 #include <CL/sycl.hpp>
 #include <iostream>
 
+// global pointer to set different attribute state for a class
+template <class T>
+struct MakeGlobalPointer {
+  typedef typename cl::sycl::global_ptr<T>::pointer_t Type;
```
Comment (keryell) [inline tests/address_spaces/address_spaces.cpp]: *NO* typedef in modern C++ !

**Comment by keryell** [comment]
A missing file perhaps?

**Comment by JoanThibault** [comment]
commit fixed

**Comment by keryell** [comment]
Making progress...
It is interesting to compare `tests/address_spaces/address_spaces_ptr.cpp` and `tests/address_spaces/address_spaces.cpp`. It would be nice to remove as much as possible from address_spaces_ptr.cpp which is not required  and already in address_spaces.cpp

**Comment by JoanThibault** [comment]
Wait before merging, I have a bug from the PSTL.

**Comment by keryell** [comment]
Thank you for your contribution.
For the bug in the SYCL PSTL, this will go in another merge request.

**Pull request merged**